pkgs/servers/search/elasticsearch/plugins.nix

```diff
@@ -36,6 +36,21 @@ in {
     };
   };
 
+  elasticsearch_cloud_aws = esPlugin rec {
+    name = "elasticsearch-cloud-aws-${version}";
+    pluginName = "elasticsearch-cloud-aws";
+    version = "2.4.2";
+    src = fetchurl {
+      url = "https://github.com/elastic/elasticsearch-cloud-aws/archive/v2.4.2.zip";
```

Comment:
The trouble with that is the elasticsearch plugin manager expects to be passed a zip file, and fetchFromGitHub always unpacks the archive after downloading it.  You could do something like this, though it's a bit hacky.

```nix
  elasticsearch_cloud_aws = esPlugin rec {
    name = "elasticsearch-cloud-aws-${version}";
    pluginName = "elasticsearch-cloud-aws";
    version = "2.4.2";
    src = fetchFromGitHub {
      owner = "elastic";
      repo = "elasticsearch-cloud-aws";
      rev = "0792440f6ff1c1a2d0002dd4497dd5a3123f867d";
      sha256 = "etc";
    };
    # fetchFromGitHub unpacks, but we want a zip. Can't reliably use fetchurl because
    # github sometimes regenrates zips and you'll get inconsistent checksums :(
    preConfigure = ''
      newsrc=$PWD/v2.4.2.zip
      (cd $src/../ && ${pkgs.zip}/bin/zip -r $newsrc $(basename $src))
      src=$newsrc
    '';
  };
```